// pluscodes/dist/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.zip = exports.flatten = exports.arrayOf = exports.parseNum = exports.valueToDigit = exports.digitToValue = exports.isValidCode = exports.isValidCoordinates = void 0;
const digits = '23456789CFGHJMPQRVWX';
const pair = `[${digits}]{2}`;
const pairOrZero = `([${digits}]|0){2}`;
const regexp = `^${pair}(${pairOrZero}){0,3}[+](${pair})?$`;
const isValidCoordinates = (subject) => {
    if (subject == null || typeof subject !== 'object')
        return false;
    return 'longitude' in subject && 'latitude' in subject;
};
exports.isValidCoordinates = isValidCoordinates;
const isValidCode = (subject) => typeof subject === 'string' && Boolean(subject.match(regexp));
exports.isValidCode = isValidCode;
const digitToValue = (x) => digits.indexOf(x);
exports.digitToValue = digitToValue;
const valueToDigit = (x) => digits.charAt(x);
exports.valueToDigit = valueToDigit;
const parseNum = (value) => typeof value === 'string' ? parseFloat(value) : value;
exports.parseNum = parseNum;
const arrayOf = (count, fill) => {
    const result = [];
    for (let i = 0; i < count; i++)
        result.push(fill);
    return result;
};
exports.arrayOf = arrayOf;
const flatten = (a) => a.reduce((acc, val) => acc.concat(val), []);
exports.flatten = flatten;
const zip = (a, b) => a.map((e, i) => [e, b[i]]);
exports.zip = zip;
